src/main/java/com/exasol/adapter/dynamodb/documentpath/DocumentPathExpression.java

```diff
@@ -0,0 +1,96 @@
+package com.exasol.adapter.dynamodb.documentpath;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Using this class a path through a document can be expressed.
+ */
+public class DocumentPathExpression {
+    private final List<PathSegment> path;
+
+    private DocumentPathExpression(final List<PathSegment> path) {
+        this.path = path;
+    }
+
+    List<PathSegment> getPath() {
+        return this.path;
+    }
+
+    /**
+     * Creates a sub path starting at startIndex until endIndex.
+     * 
+     * @param startIndex index in path for new path to start
+     * @param endIndex   index in path for new path to end
+     * @return new {@link DocumentPathExpression}
+     */
+    public DocumentPathExpression getSubPath(final int startIndex, final int endIndex) {
+        return new DocumentPathExpression(Collections.unmodifiableList(this.path.subList(startIndex, endIndex)));
+    }
+
+    /**
+     * Gives the size of this path expression.
+     * 
+     * @return size
+     */
+    public int size() {
+        return this.path.size();
+    }
+
+    /**
+     * Builder for {@link DocumentPathExpression}.
+     */
+    public static class Builder {
+        private final List<PathSegment> path = new ArrayList<>();
+
+        /**
+         * Appends a {@link PathSegment} to the current path.
+         * 
+         * @param segment path segment to append
+         * @return this for as fluent programming interface
```

Comment:
```suggestion
         * @return <code>this</code> instance for fluent programming
```

Check other places with the same comment too